### PR TITLE
Fix `--generate-mono-glue` printing an unknown CLI argument warning

### LINKED ISSUE #99069: Fail to build mono glue
### Tested versions

- Tested in 0f5f3bc9546b46b2029fc8896dc859697f1eab97

### System information

Windows 11 Pro

### Issue description

After the latest commits on godot (it used to build fine before that) the mono glue fails to be built with the following errors:
` Godot.SourceGenerators.Internal succeeded (1.3s) → glue\GodotSharp\Godot.SourceGenerators.Internal\bin\Debug\netstandard2.0\Godot.SourceGenerators.Internal.dll
  GodotSharp failed with 3 error(s) and 7 warning(s) (6.2s)
    D:\Source\Other\gd-engine\modules\mono\glue\GodotSharp\GodotSharp\Compat.cs(35,12): error CS0246: The type or namespace name 'AnimationMethodCallMode' could not be found (are you missing a using directive or an assembly reference?)
    D:\Source\Other\gd-engine\modules\mono\glue\GodotSharp\GodotSharp\Compat.cs(51,12): error CS0246: The type or namespace name 'AnimationProcessCallback' could not be found (are you missing a using directive or an assembly reference?)
    D:\Source\Other\gd-engine\modules\mono\glue\GodotSharp\GodotSharp\Compat.cs(62,12): error CS0246: The type or namespace name 'AnimationProcessCallback' could not be found (are you missing a using directive or an assembly reference?)`
This is most surely caused by the new errors that popped in during the glue generation procedure which  are all as this following one:
`Cannot reference editor type 'EditorImportPlugin' in documentation for core type 'ImageTexture'
Generating ImageTexture3D.cs...
Generating ImageTextureLayered.cs...
ERROR: Cannot resolve method reference in documentation: 'Image.create'.
   at: BindingsGenerator::_append_xml_method (modules\mono\editor\bindings_generator.cpp:1151)`
This does happen quite a few times on other areas, like:
`ERROR: Cannot resolve method reference in documentation: 'Skeleton3D.clear_bones_global_pose_override'.
   at: BindingsGenerator::_append_xml_method (modules\mono\editor\bindings_generator.cpp:1151)`
and
`ERROR: Cannot resolve method reference in documentation: 'make_polygons_from_outlines'.
   at: BindingsGenerator::_append_xml_method (modules\mono\editor\bindings_generator.cpp:1151)
ERROR: Cannot resolve method reference in documentation: 'make_polygons_from_outlines'.
   at: BindingsGenerator::_append_xml_method (modules\mono\editor\bindings_generator.cpp:1151)
ERROR: Cannot resolve method reference in documentation: 'make_polygons_from_outlines'.
   at: BindingsGenerator::_append_xml_method (modules\mono\editor\bindings_generator.cpp:1151)`

### Steps to reproduce

Simply use latest commit to build from source with mono support, it will build the engine but fail to generate/build the mono glue.

### Minimal reproduction project (MRP)

Build issue.

### PR DESCRIPTION
**Pull request by Calinou**
- Follow-up to https://github.com/godotengine/godot/pull/98895.

___

- Abort with an error when using `--generate-mono-glue` on builds without C# support (similar to `--test` on builds with tests disabled).
- Fix unknown CLI argument warning messages not being visible on Windows.
- Fix `--test` error message not being visible on Windows on builds with tests disabled.

The issue about prints not being visible is due to how `ERR_PRINT()` and `WARN_PRINT()` work. We don't need their stack trace to be visible here anyway. We can reinstate the coloring later in a future PR (which will apply to all CLI arguments by adding a new method for it).

- This closes https://github.com/godotengine/godot/issues/99069.

### CODE REVIEW

**Review by raulsntos — APPROVED** [review]
Thanks! I can confirm this fixes the `--generate-mono-glue` issue.

File: main/main.cpp [hunk 1]
```diff
@@ -1801,12 +1801,22 @@ Error Main::setup(const char *execpath, int argc, char *argv[], bool p_second_ph
...
+			// Handle arguments specific to C#-enabled builds.
+			const bool using_mono_args = arg == "--generate-mono-glue";
+#ifndef MODULE_MONO_ENABLED
+			if (using_mono_args) {
+				OS::get_singleton()->print(
+						"`--generate-mono-glue` was specified on the command line, but this Godot binary was compiled without .NET support. Aborting.\n"
+						"To be able to generate C# glue code, use the `module_mono_enabled=yes` SCons option when compiling Godot.\n");
+				goto error;
+			}
+#endif
```
Comment (akien-mga) [inline main/main.cpp:1813]: We're starting to accumulate a lot of hacks in `main.cpp` for the Mono module... I'm not fond of increasing the tech debt further here.

There's also `--class-db-json` in the Mono module that's not being handled here.

Comment (akien-mga) [reply main/main.cpp:1813]: See also #99224.

Comment (raulsntos) [reply main/main.cpp:1813]: > There's also `--class-db-json` in the Mono module that's not being handled here.

I don't think anybody's using this, it's an undocumented option and could probably be replaced with `--dump-extension-api`.

**Comment by akien-mga** [comment]
Superseded by #99300.

**Pull request closed without merging**